libs/ShaderProgram/ShaderProgram.cpp

```diff
@@ -0,0 +1,184 @@
+#include "ShaderProgram.h"
+#include <fstream>
+#include <string>
+
+ShaderProgram::ShaderProgram() :progarmId(0)
+{
+}
+
+void ShaderProgram::LoadShaders(const char * vertex_filenname, const char * fragment_filename)
```

Comment:
Some functions use `std::string` others ` char *` ... may be good to just stick to one `std::string`?

Reply:
The public method has been changed to `std::string&`. The private method still uses `const char *` due to the OpenGL API functions used this data type to compile and link shaders.